File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/TestResultPollingToken.java
package uk.nhs.nhsx.virology;

import uk.nhs.nhsx.core.ValueType;

import static uk.nhs.nhsx.core.Preconditions.checkArgument;

public class TestResultPollingToken extends ValueType<TestResultPollingToken> {

    private TestResultPollingToken(String value) {
        super(value);
        checkArgument(!value.isEmpty(), "Test result polling token not present");
    }

    public static TestResultPollingToken of(String value) {
        return new TestResultPollingToken(value);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/TestResultPollingToken.kt
package uk.nhs.nhsx.virology

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class TestResultPollingToken private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<TestResultPollingToken>(::TestResultPollingToken)
}
